tests/distributions/test_gaussian_mixtures.py

```diff
@@ -0,0 +1,108 @@
+from __future__ import absolute_import, division, print_function
+
+import math
+
+import torch
+
+import pytest
+from pyro.distributions import MixtureOfDiagNormalsSharedCovariance
+from tests.common import assert_equal
+
+
+@pytest.mark.parametrize('mix_dist', [MixtureOfDiagNormalsSharedCovariance])
+@pytest.mark.parametrize('K', [3])
+@pytest.mark.parametrize('D', [2, 4])
+@pytest.mark.parametrize('batch_mode', [True, False])
```

Comment:
Add a test for `len(sample_shape) > 1`, e.g. via
```py
@pytest.mark.parametrize('sample_shape', [(), (n_samples,), (2, n_samples)])
```

Reply:
i've run these tests before but think it's too much for travis